### PR TITLE
doc: Update 0.19.0 release notes

### PR DESCRIPTION
**Pull request by kaiyux**
# PR title

Please write the PR title by following template:

[JIRA ticket link/nvbug link/github issue link][fix/feat/doc/infra/...] \<summary of this PR\>

For example, assume I have a PR hope to support a new feature about cache manager of Jira TRTLLM-1000 ticket, it would be like

[TRTLLM-1000][feat] Support a new feature about cache manager

## Description

Please explain the issue and the solution in short.

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--disable-fail-fast --skip-test --stage-list "A10-1, xxx" --gpu-type "A30, H100_PCIe" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-[Post-Merge]-1, xxx"]`

Launch build/test pipelines. All previously running jobs will be killed.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests. Will also run L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-[Post-Merge]-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-[Post-Merge]-1, xxx".

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

### CODE REVIEW

**Review by litaotju — APPROVED** [review]

File: docs/source/release-notes.md [hunk 1]
```diff
@@ -5,6 +5,122 @@
...
+### Known Issues
+
+### Fixed Issues
+  - Fixed misused length argument of PluginField. Thanks to the contribution from @jl749 in #2712. This also fixes #2685.
+  - Fixed a Llama-3.2 SmoothQuant convert checkpoint issue. (#2677)
+  - Fixed a bug when loading an engine using LoRA through the LLM API. (#2782)
+  - Fixed incorrect batch slot usage in `addCumLogProbs` kernel. Thanks to the contribution from @aotman in #2787.
+  - Fixed incorrect output for Llama-3.2-11B-Vision-Instruct. (#2796)
+
+### Infrastructure Changes
```
Comment (chzblych) [inline docs/source/release-notes.md:121]:
```suggestion
### Infrastructure Changes
  - The dependent NVIDIA ModelOpt version is updated to 0.27.
```
```

File: docs/source/release-notes.md [hunk 2]
```diff
@@ -5,6 +5,122 @@
...
+  - Redesigned the multi‑stream API for DeepSeek
+
+### Known Issues
+
+### Fixed Issues
+  - Fixed misused length argument of PluginField. Thanks to the contribution from @jl749 in #2712. This also fixes #2685.
+  - Fixed a Llama-3.2 SmoothQuant convert checkpoint issue. (#2677)
+  - Fixed a bug when loading an engine using LoRA through the LLM API. (#2782)
+  - Fixed incorrect batch slot usage in `addCumLogProbs` kernel. Thanks to the contribution from @aotman in #2787.
+  - Fixed incorrect output for Llama-3.2-11B-Vision-Instruct. (#2796)
```
Comment (chzblych) [inline docs/source/release-notes.md:119]:
```suggestion
  - Fixed incorrect output for Llama-3.2-11B-Vision-Instruct. (#2796)
  - Removed the necessary of `--extra-index-url https://pypi.nvidia.com` when running `pip install tensorrt-llm`.
```

File: docs/source/release-notes.md [hunk 3]
```diff
@@ -5,6 +5,122 @@
...
+  - Exposed `kc_cache_retention_config` from C++ `executor` API to the LLM API.
+  - Moved `BuildConfig` arguments to `LlmArgs`.
+  - Removed speculative decoding parameters from stateful decoders
+  - Exposed `DecoderState` via bindings and integrated it in decoder
+  - Refactored the `LlmArgs` with `Pydantic` and migrated remaining pybinding configurations to Python
+  - Refactored disaggregated serving scripts
+  - Added `numNodes` to `ParallelConfig`
+  - Redesigned the multi‑stream API for DeepSeek
+
+### Known Issues
```
Comment (chzblych) [inline docs/source/release-notes.md:112]:
```suggestion
### Known Issues
  - The PyTorch workflow on SBSA is incompatible with bare metal environments like Ubuntu 24.04. Please use the [PyTorch NGC Container](https://catalog.ngc.nvidia.com/orgs/nvidia/containers/pytorch) for optimal support on SBSA platforms.
```

**Review by chzblych — APPROVED** [review]

File: docs/source/release-notes.md [hunk 4]
```diff
@@ -5,6 +5,127 @@
...
+    - Added Qwen2 MoE support.
+    - Applied `AutoTuner` to both Fused MoE and NVFP4 Linear operators.
+    - Introduced a `UserBuffers` allocator.
+    - Added Deepseek eager mode AllReduce fusion support.
+    - Added Multi-Token Prediction (MTP) support. Refer to the “Multi-Token Prediction (MTP)” section of `examples/deepseek_v3/README.md`.
+    - Added FlashMLA support for SM90.
+    - Added support for enabling MTP with CUDA graph padding.
+    - Added initial EAGLE-3 implementation.
+    - Added support for FP8 MLA on NVIDIA Hopper and Blackwell GPUs.
+  - **AutoDeploy workflow**.
```
Comment (lucaslie) [inline docs/source/release-notes.md:43]: @laikhtewari, do you have any thoughts on calling this `AutoDeploy workflow`. Just confirming since we had a few discussions about how we would position autodeploy compared to the manual pytorch workflow? 

We could name it something like `AutoDeploy for PyTorch workflow`

cc @suyoggupta

Comment (laikhtewari) [reply docs/source/release-notes.md:43]: I wouldn't call it a workflow, maybe just feature?

From a deployer perspective, the workflow is basically the same to run models with explicit trtllm definitions vs through AD

From model author perspective, there is a workflow for explicit definition and there's an absence of a workflow for AD

I also wouldn't reference the PyTorch architecture because once we transition over, it will be *the* architecture of TensorRT LLM, so calling it out explicitly might be confusing

Comment (kaiyux) [reply docs/source/release-notes.md:43]: Thanks for the discussion, does `AutoDeploy for PyTorch workflow` as mentioned by @lucaslie work?

File: docs/source/release-notes.md [hunk 5]
```diff
@@ -5,6 +5,127 @@
...
+  - Added `numNodes` to `ParallelConfig`.
+  - Redesigned the multi‑stream API for DeepSeek.
+
+### Fixed Issues
+  - Fixed misused length argument of PluginField. Thanks to the contribution from @jl749 in #2712. This also fixes #2685.
+  - Fixed a Llama-3.2 SmoothQuant convert checkpoint issue. (#2677)
+  - Fixed a bug when loading an engine using LoRA through the LLM API. (#2782)
+  - Fixed incorrect batch slot usage in `addCumLogProbs` kernel. Thanks to the contribution from @aotman in #2787.
+  - Fixed incorrect output for Llama-3.2-11B-Vision-Instruct. (#2796)
+  - Removed the necessary of `--extra-index-url https://pypi.nvidia.com` when running `pip install tensorrt-llm`.
```
Comment (a-mccarthy) [inline docs/source/release-notes.md:115]:
```suggestion
  - Including `--extra-index-url https://pypi.nvidia.com` is no longer required when running `pip install tensorrt-llm`.
```

Was this requirement ever noted in the docs? I see its listed as a known issue in the .17 release notes, was the third party dependency removed as well?

File: docs/source/release-notes.md [hunk 6]
```diff
@@ -5,6 +5,127 @@
 All published functionality in the Release Notes has been fully tested and verified with known limitations documented. To share feedback about this release, access our [NVIDIA Developer Forum](https://forums.developer.nvidia.com/).
 
 
+## TensorRT-LLM Release 0.19.0
+
+### Key Features and Enhancements
+  - **The C++ runtime is now open sourced.**
+  - **PyTorch workflow**
+    - Added DeepSeek V3/R1 support. Refer to `examples/deepseek_v3/README.md`, also to the blog `docs/source/blogs/Best_perf_practice_on_DeepSeek-R1_in_TensorRT-LLM.md`.
```
Comment (a-mccarthy) [inline docs/source/release-notes.md:13]: Will this blog be added to the TOC? I see that its in the [repo](https://github.com/NVIDIA/TensorRT-LLM/blob/main/docs/source/blogs/Best_perf_practice_on_DeepSeek-R1_in_TensorRT-LLM.md), but its not visible on the side navigation yet. It might be nice to add a link to this page directly.

File: docs/source/release-notes.md (hunk 4 above)
Comment (lucaslie) [reply docs/source/release-notes.md:43]: I vote for `AutoDeploy for PyTorch workflow` for now. This way we don't position it as new workflow but rather but part of the pytorch workflow but also clearly position it as separate feature

File: docs/source/release-notes.md (hunk 5 above)
Comment (kaiyux) [reply docs/source/release-notes.md:115]: Yes it was noted in the docs, see this page for example: https://nvidia.github.io/TensorRT-LLM/0.18.2/installation/linux.html

File: docs/source/release-notes.md (hunk 6 above)
Comment (kaiyux) [reply docs/source/release-notes.md:13]: What's TOC?

File: docs/source/release-notes.md (hunk 4 above)
Comment (kaiyux) [reply docs/source/release-notes.md]: Updated as `AutoDeploy for PyTorch workflow`, thanks.

**Review by chzblych — APPROVED** [review]

File: docs/source/release-notes.md [hunk 7]
```diff
@@ -5,6 +5,127 @@
...
+  - **AutoDeploy for PyTorch workflow**.
+    - The AutoDeploy for PyTorch workflow is an **experimental** feature in `tensorrt_llm._torch.auto_deploy`.
+    - AutoDeploy provides an automated path from off-the-shelf models to optimized deployment in the TensorRT-LLM runtime.
+    - Check out `examples/auto_deploy/README.md` for more details.
+  - LLM API
+    - Added batched logits processor support.
+    - Added EAGLE support.
+    - Added abort request support.
+    - Added `get_stats` support.
+    - [BREAKING CHANGE] Added dynamic logits processor support, and deprecated static logits processor.
```
Comment (Superjomn) [inline docs/source/release-notes.md:52]: - Added multi-node on Slurm-like clusters, refer to `examples/llm-api/llm_mgmn_*.sh` for examples

Comment (kaiyux) [reply docs/source/release-notes.md]: Updated with minor modification, thanks.

File: docs/source/release-notes.md (hunk 6 above)
Comment (kaiyux) [reply docs/source/release-notes.md:13]: @a-mccarthy I'm merging this PR now, for any further concerns please directly let me know. Thanks.
